src/vm/threadsuspend.cpp

```diff
@@ -2545,7 +2545,7 @@ void Thread::RareDisablePreemptiveGC()
     // block until resume
     if (((GCHeapUtilities::IsGCInProgress()  && (this != ThreadSuspend::GetSuspensionThread())) ||
         (m_State & (TS_UserSuspendPending | TS_DebugSuspendPending | TS_StackCrawlNeeded))) &&
-        (!g_fSuspendOnShutdown || IsFinalizerThread() || IsShutdownSpecialThread()))
+        (IsFinalizerThread() || IsShutdownSpecialThread()))
```

Comment:
Should any thread other than the suspension thread be able to go into this block, and not just the finalizer / shutdown-special thread?

Reply:
You are right. I have looked at `g_fSuspendOnShutdown` backwards.